mlfinlab/tests/test_correlation_driven_nonparametric_learning_k.py

```diff
@@ -0,0 +1,114 @@
+"""
+Tests Correlation Driven Nonparametric Learning - K.
+"""
+from unittest import TestCase
+import os
+import numpy as np
+import pandas as pd
+from mlfinlab.online_portfolio_selection.pattern_matching.correlation_driven_nonparametric_learning_k import CorrelationDrivenNonparametricLearningK
+
+
+class TestCorrelationDrivenNonparametricLearningK(TestCase):
+    # pylint: disable=too-many-public-methods
+    # pylint: disable=unsubscriptable-object
+    """
+    Tests different functions of the Correlation Driven Nonparametric Learning - K class.
+    """
+
+    def setUp(self):
+        """
+        Sets the file path for the tick data csv.
+        """
+        # Set project path to current directory.
+        project_path = os.path.dirname(__file__)
+        # Add new data path to match stock_prices.csv data.
+        data_path = project_path + '/test_data/stock_prices.csv'
+        # Read csv, parse dates, and drop NaN.
+        self.data = pd.read_csv(data_path, parse_dates=True, index_col="Date").dropna(axis=1)
+
+    def test_corn_k_solution(self):
+        """
+        Test the calculation of CORN-K.
+        """
+        # Initialize CORN-K.
+        corn_k = CorrelationDrivenNonparametricLearningK(window=2, rho=2, k=1)
+        # Allocates asset prices to CORN-K.
+        corn_k.allocate(self.data, resample_by='M')
+        # Create np.array of all_weights.
+        all_weights = np.array(corn_k.all_weights)
+        # Check if all weights sum to 1.
+        for i in range(all_weights.shape[0]):
+            weights = all_weights[i]
+            assert (weights >= 0).all()
+            assert len(weights) == self.data.shape[1]
+            np.testing.assert_almost_equal(np.sum(weights), 1)
+
+    def test_corn_k_window_error(self):
+        """
+        Tests ValueError if window is not an integer.
```

Comment:
Consider testing ValueErrors all in one function and thus eliminating the `too-many-public-methods` pylint error

(Here and in the unit tests below)